### PR TITLE
Validates permission before calling beforeSave trigger

### PR DESCRIPTION
**Pull request by davimacedo**
Solves issue https://github.com/parse-community/parse-server/issues/5513

### CODE REVIEW

File: src/RestWrite.js [hunk 1]
```diff
@@ -220,6 +220,34 @@ RestWrite.prototype.runBeforeSaveTrigger = function() {
   }
 
   return Promise.resolve()
+    .then(() => {
+      let databasePromise = null;
```
Comment (acinader) [inline src/RestWrite.js]: can you add a comment here, please.

File: src/Controllers/DatabaseController.js [hunk 2]
```diff
@@ -831,13 +849,19 @@ class DatabaseController {
           .then(schema => {
             transformAuthData(className, object, schema);
             flattenUpdateOperatorsForCreate(object);
+            if (validateOnly) {
+              return object;
```
Comment (acinader) [inline src/Controllers/DatabaseController.js]: what is getting returned here?

File: src/Controllers/DatabaseController.js [hunk 3]
```diff
@@ -588,6 +602,9 @@ class DatabaseController {
               'Object not found.'
             );
           }
+          if (validateOnly) {
+            return result;
```
Comment (acinader) [inline src/Controllers/DatabaseController.js]: what will `result` be in a `validateOnly` case?

File: src/RestWrite.js (hunk 1 above)
Comment (davimacedo) [reply src/RestWrite.js]: Just done

File: src/Controllers/DatabaseController.js (hunk 3 above)
Comment (davimacedo) [reply src/Controllers/DatabaseController.js]: In both cases (create and update), it is returning the original object.

File: src/Controllers/DatabaseController.js (hunk 2 above)
Comment (davimacedo) [reply src/Controllers/DatabaseController.js]: The original object that would go to the database but in the case of validateOnly, it is just returned after the validations.

**Pull request merged**